tests/integration/targets/dest_file_module_helper/tasks/main.yml

```diff
@@ -0,0 +1,433 @@
+---
+####################################################################
+# WARNING: These are designed specifically for Ansible tests       #
+# and should not be used as examples of how to write Ansible roles #
+####################################################################
+
+# (c) 2022 DEMAREST Maxime <maxime@indelog.fr>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: set dest_dir and dest_file
+  set_fact:
+    dest_dir: "{{ remote_tmp_dir }}/destdir/"
+    dest_file: "{{ remote_tmp_dir }}/destdir/destfile.txt"
+
+- name: 001 fail when dest_file not exists
+  block:
+    - name: 001 create writeable and readable dest_dir
+      ansible.builtin.file:
+        path: "{{ dest_dir }}"
+        mode: "0777"
+        state: "directory"
+    - name: 001 run fail when dest_file not exists with create false
+      dummy_module:
+        path: "{{ dest_file }}"
+        value: "This is dummy content."
+        allow_creation: false
+      ignore_errors: true
+      register: last_result
+    - name: 001 assert fail when dest_file not exists
+      ansible.builtin.assert:
+        that:
+          - last_result.failed == true
+          - last_result.changed == false
+          - last_result.msg == "{{ dest_file }} does not not exist !"
+    - name: 001 clean dest_dir
+      ansible.builtin.file:
+        path: "{{ dest_dir }}"
+        state: "absent"
+
+- name: 002 fail when dest_file not exists and create on non writeable parent
+  block:
+    - name: 002 create readonly dest_dir
+      ansible.builtin.file:
+        path: "{{ dest_dir }}"
+        mode: "0555"
+        state: "directory"
+    - name: 002 add read right on remote temp dir to nobody
+      ansible.builtin.file:
+        path: "{{ remote_tmp_dir }}"
+        mode: "0755"
+    - name: 002 run fail when dest_file not exists and create on non writeable parent
+      dummy_module:
+        path: "{{ dest_file }}"
+        value: "Content A"
+        allow_creation: true
+      ignore_errors: true
+      become: "{{(ansible_user_uid == 0) | ternary(true, false)}}"
```

Comment:
The purpose here is not to test with both value, but is to adapt the behaviour of the test depending the user that run it :

 * If we run test as `root` (is the case with Docker container), i can't test the case when the file is not readable or writeable (because `root` always can access to the file). So to do this test with `root` it is necessary to run it with other user.
 * But if the testing user is not root (when i want to do quick run of the test, i do not run them with Docker, so they are running by my current user), i can't switch user. So we do not do it.

 I'm not satisfied by this, this very depending to the platform that run the test (i use the `nobody` user) but i not found a mean to do this in other way.

 I'il replace `{{ ansible_user_uid == 0 | ternary(true, false) }}` by `{{ ansible_user_uid == 0}}`.